server/services/db_service.go

```diff
@@ -0,0 +1,14 @@
+package services
+
+import (
+	"errors"
+	"github.com/go-sql-driver/mysql"
+)
+
+func IsDuplicatedKeyError(err error) bool {
+	var mysqlError *mysql.MySQLError
+	if errors.As(err, &mysqlError) {
+		return mysqlError.Number == 1062
```

Comment:
Because I can't understand what this number means alone, I would put it in a variable that explains what it means